lib/internal/test_runner/tap_checker.js

```diff
@@ -0,0 +1,128 @@
+'use strict';
+
+const { NumberParseInt } = primordials;
+const {
+  codes: { ERR_TAP_VALIDATOR_ERROR },
+} = require('internal/errors');
+
+class TAPValidationStrategy {
```

Comment:
same here - I don't think we should validate TAP - since we treat anything we cannot parse as a comment, so why throw?

Reply:
The check/validation process is run independently from the parsing. Consumers need to call `parser.check()` after the parsing process.

Reply:
@cjihrig  Can I resolve this converstion?

Reply:
Sure. I'll just do that while I'm here.